packages/core/src/components/toast/toast.md

```diff
@@ -58,51 +51,71 @@ OurToaster.update(key, { message: "Still toasted!" });
-given `container`. The `container` determines which element toasts are positioned relative to; the
-default value of `<body>` allows them to use the entire viewport.
+The `Toaster` will be rendered into a new element appended to the given `container`. The `container` determines which element toasts are positioned relative to; the default value of `<body>` allows them to use the entire viewport.
 
 Note that the return type is `IToaster`, which is a minimal interface that exposes only the instance
 methods detailed below. It can be thought of as `Toaster` minus the `React.Component` methods,
 because the `Toaster` should not be treated as a normal React component.
 
-@interface IToasterProps
+Your application can contain several `Toaster` instances and easily share them across the codebase as modules.
+
+The following code samples demonstrate our preferred pattern for intergrating a toaster into a React application:
+
+#### `toaster.ts`
+```tsx
+import { Position, Toaster } from "@blueprintjs/core";
 
-@### Instance methods
+/** Singleton toaster instance. Create separate instances for different options. */
+export const AppToaster = Toaster.create({
+    className: "recipe-toaster",
+    position: Position.TOP,
+});
+```
 
-<div class="docs-interface-name">IToaster</div>
+#### `application.ts`
+```tsx
+import { Button } from "@blueprintjs/core";
+import { AppToaster } from "./toaster";
 
-- `show(props: IToastProps): string` — Show a new toast to the user.
-Returns the unique key of the new toast.
-- `update(key: string, props: IToastProps): void` —
-Updates the toast with the given key to use the new props.
-Updating a key that does not exist is effectively a no-op.
-- `dismiss(key: string): void` — Dismiss the given toast instantly.
-- `clear(): void` — Dismiss all toasts instantly.
-- `getToasts(): IToastProps[]` — Returns the options for all current toasts.
+export class App extends React.PureComponent {
+    render() {
+        return <Button onClick={this.showToast} text="Toast please" />;
+    }
 
-@interface IToastProps
+    showToast = () => {
+        // create toasts in response to interactions.
+        // in most cases, it's enough to simply create and forget (thanks to timeout).
+        AppToaster.show({ message: "Toasted." });
+    }
+}
+```
 
-@### React component
+@## React component usage
 
-The `Toaster` React component is a stateful container for a single list of toasts. Internally, it
-uses [`Overlay`](#core/components/overlay) to manage children and transitions. It can be vertically
-aligned along the top or bottom edge of its container (new toasts will slide in from that edge) and
-horizontally aligned along the left edge, center, or right edge of its container.
+Render the `<Toaster>` component like any other element. Then, either access the instance methods by supplying
+a `ref` handler (see example below), or supply rendered `<Toast>` elements as `children` of the `<Toaster>`.
```

Comment:
I'm not sure why we're suggesting a `ref` handler... if someone really wanted to do this in a React-idiomatic way, wouldn't they provide toasts as `children`? What are the benefits of this over the `Toaster.create()` API? I feel all these options make the usage docs more confusing.

Reply:
i'll move the `children` suggestion first and note that you can `ref` but please don't (see next paragraph).